### DISCUSSION TITLE
Share what you (plan to) read with a URL

### DISCUSSION THREAD

**Discussion by soc1ety** [mateusz-bak/openreads#699] [Ideas]
**Is your feature request related to a problem? Please describe.**

On similar services like Babelio we can share a list of books by generating a custom URL that anyone can see given he has already the URL. 
It would be cool if we could do the same on OpenReads, in case we want to make some recommendations without giving the books' title one by one.

**Describe the solution you'd like**

A functionality that allows to create a link from a list of read / dropped / plan-to-read books with anyone

**Describe alternatives you've considered**

Maybe the possibility to import books from someone's list thereafter ? 

As I've never done any mobile development, I don't know at all if it's possible to implement something like that, but here's my request !
Keep up the good work with the actual application mateusz, your work is highly appreciated.

**Comment by mateusz-bak** [comment]
Hello @soc1ety thank you for the suggestion although for having reading information on a link we would have to host a backend and for now the application is designed to work offline (except Open Library API). Maybe other solution would be to share books in a form of text not a URL?

**Comment by soc1ety** [comment]
Oh I didn't know about the backend thing, thanks for telling me.

Otherwise yeah it'd be a pretty good solution for now I guess, sharing readings would be a lot easier aswell

**Comment by Douile** [comment]
It would be possible to have sharable links without a backend.

To share individual books we could generate links to openlibrary or goodreads (probably not good reads). By adding an intent filter that handles the URL openlibrary users could automatically open the app when receiving said URLs.

To share lists of books you would probably need a dummy site (any URL, preferably one we control like a free github pages site), all the book information could be added to the URL in various different ways. Then by adding an intent filter for said URL app users can receive the URL and decode the book list information. This could be made more friendly to non-app users by hosting some static javascript on the site that decodes the book info and displays it in the page.

The downside to these approaches is that if the links are visited the information is shared with a third party which could make a simple text share preferable.

We could also use `app://` links which can never be associated with a site.

I would happy to help implementing this if it is something people are interested in.

[Documentation on adding link handling](https://developer.android.com/training/app-links/deep-linking)

**Comment by apobrt** [comment]
I would love to see this feature implemented!
I personally find that both options could be useful:
- export the list of books to be read as plain text (easily pasted anywhere), also useful for those who don't use and don't want to use Openreads
- Export a link : I like the idea of using app://

**Comment by mateusz-bak** [comment]
Hello, there is a plan to create Openreads Hub - optional, and private platform books tracking with a self hosted option. When the plan is more developed I will share it and create a central discussion. For now I'm adding a `hub`  label so we can track proposed features.

When Hub is created the link could be shared easily.
